SecurityPkg/VariableAuthenticated/SecureBootConfigDxe/SecureBootConfigImpl.c

```diff
@@ -2649,22 +2652,59 @@ UpdateDeletePage (
                                     + sizeof (EFI_SIGNATURE_LIST)
                                     + CertList->SignatureHeaderSize
                                     + Index * CertList->SignatureSize);
+
+      // Get the required buffer size
+      X509GetCommonName (
+        (UINT8*)Cert->SignatureData,
+        (UINTN)CertList->SignatureSize,
+        NULL,
+        &CertificateInfoStrSize
+        );
+
+      CertificateInfoStr8 = AllocateZeroPool (CertificateInfoStrSize * sizeof(CHAR8));
+      if (CertificateInfoStr8 == NULL) {
+        Status = EFI_OUT_OF_RESOURCES;
+        goto ON_EXIT;
+      }
+      CertificateInfoStr = AllocateZeroPool (CertificateInfoStrSize * sizeof(CHAR16));
+      if (CertificateInfoStr == NULL) {
+        Status = EFI_OUT_OF_RESOURCES;
+        goto ON_EXIT;
+      }
+
       //
       // Display GUID and help
       //
-      GuidToString (&Cert->SignatureOwner, GuidStr, 100);
-      GuidID = HiiSetString (PrivateData->HiiHandle, 0, GuidStr, NULL);
+      RStatus = X509GetCommonName (
+        (UINT8*)Cert->SignatureData,
+        (UINTN)CertList->SignatureSize,
+        CertificateInfoStr8,
+        &CertificateInfoStrSize
+        );
+
+      if (!EFI_ERROR (RStatus)) {
+        AsciiStrToUnicodeStrS (CertificateInfoStr8, CertificateInfoStr, CertificateInfoStrSize);
+        CertificateInfoID  = HiiSetString (PrivateData->HiiHandle, 0, CertificateInfoStr, NULL);
+      } else {
+        CertificateInfoID  = HiiSetString (PrivateData->HiiHandle, 0, UNKNOWN_CERT, NULL);
+      }
+
       HiiCreateCheckBoxOpCode (
         StartOpCodeHandle,
         (EFI_QUESTION_ID)(QuestionIdBase + GuidIndex++),
         0,
         0,
-        GuidID,
+        CertificateInfoID,
         Help,
         EFI_IFR_FLAG_CALLBACK | EFI_IFR_FLAG_RESET_REQUIRED,
         0,
         NULL
         );
+
+      FreePool (CertificateInfoStr8);
+      FreePool (CertificateInfoStr);
+      CertificateInfoStr8 = NULL;
+      CertificateInfoStr = NULL;

```

Comment:
```suggestion
      FreePool (CertificateInfoStr8);
      FreePool (CertificateInfoStr);
```

Just FreePool is enough. In the next for loop iteration the AllocateZeroPool will set these pointers either to a new buffer or NULL if cannot allocate memory. No need to do it explicitly.

Reply:
@miczyg1 That's exactly what I thought, but as soon as I remove the ` = NULL` lines and rebuild I get a `DXE_ASSERT` entering the Delete Signature menus, 100% reproducibility:

```
DXE_ASSERT!: [DxeCore] /home/coreboot/coreboot/payloads/external/edk2/workspace/Dasharo/MdeModulePkg/Core/Dxe/Mem/Pool.c (710): Head->S
ignature == ((('p') | ('h' << 8)) | ((('d') | ('0' << 8)) << 16)) || Head->Signature == ((('p') | ('h' << 8)) | ((('d'
ASSERT_EFI_ERROR (Status = Invalid Parameter)
```

Reply:
Which delete signature? DB, DBX, KEK?

It might be a sign that something may be wrong elsewhere

Reply:
KEK and DB, DBX isn't affected, probably since it doesn't display the Common Names anywhere

Reply:
> KEK and DB, DBX isn't affected, probably since it doesn't display the Common Names anywhere

What are you talking? The whole PR is supposed to display common name/issuer for every possible certificate. You even attached pictures with the human readable certs in DB and KEK delete options...

Reply:
I took a deeper look and now understand why we have a NULL assignment here. Before we exit the FOR loop we end up freeing the CertificateInfoStr8 and CertificateInfoStr, but it doesn't meean these pointers are set to NULL. SO when we exit the for loop we reach ON_EXIT label, which will try to free  CertificateInfoStr8 and CertificateInfoStr a second time. This second free is causing the assert. Closing this comment.